modules/python/test/test_misc.py

```diff
@@ -50,39 +68,312 @@ class Arguments(NewOpenCVTests):
+
+        a = np.array([1, 2, 0], dtype=np.bool)
+        for i in range(len(a)):
+            expected = 'Bool: {}'.format(bool(a[i])).lower()
+            actual = cv.utils.dumpBool(a[i]).lower()
+            self.assertEqual(expected, actual,
+                             msg="Can't convert array element {0} of type {1}\n"
+                             "Expected: {2} vs Actual: {3}".format(
+                                 a[i],
+                                 type(a[i]).__name__,
+                                 expected, actual
+                             ))
+
+        with self.assertRaises(
+            TypeError,
+            msg='Exception is not risen for tuple slice with 1 element'
+        ):
+            _ = cv.utils.dumpBool((True, False, False)[0:1])
+
+        for not_convertible in (1.2, np.float(2.3), -123, 's', 'str', (1, 2),
+                                [1, 2], a):
+            with self.assertRaises(
+                (TypeError, OverflowError),
+                msg='Exception is not risen for {} of type {}'.format(
+                    not_convertible,
+                    type(not_convertible).__name__
+                )
+            ):
+                _ = cv.utils.dumpBool(not_convertible)
+
+    def test_parse_to_int(self):
+        min_int, max_int = get_limits(ctypes.c_int)
+
+        for convertible in (-10, -1, 2, True, False, int(43.2), np.uint8(15),
+                            np.int8(33), np.int16(-13), np.int32(4),
+                            np.int64(345), (23), min_int, max_int):
+            expected = 'Int: {0:d}'.format(convertible).lower()
+            actual = cv.utils.dumpInt(convertible).lower()
+            self.assertEqual(expected, actual,
+                             msg="Can't convert {0} of type {1}\n"
+                             "Expected: {2} vs Actual: {3}".format(
+                                 convertible,
+                                 type(convertible).__name__,
+                                 expected, actual
+                             ))
+
+        a = np.array([1, 2, 0], dtype=np.int)
+        for i in range(len(a)):
+            expected = 'Int: {0:d}'.format(a[i]).lower()
+            actual = cv.utils.dumpInt(a[i]).lower()
+            self.assertEqual(expected, actual,
+                             msg="Can't convert array element {0} of type {1}\n"
+                             "Expected: {2} vs Actual: {3}".format(
+                                 a[i],
+                                 type(a[i]).__name__,
+                                 expected, actual
+                             ))
+
+        with self.assertRaises(
+            TypeError,
+            msg='Exception is not risen for tuple slice with 1 element'
+        ):
+            _ = cv.utils.dumpInt((1, 2, 12)[0:1])
+
+        for not_convertible in (1.2, np.float(4), float(3), np.double(45), 's',
+                                'str', np.array([1, 2]), (1,), [1, 2],
```

Comment:
It make sense to add these cases too:
```
#np.float32(5),
np.float64(6),
#np.array([11], dtype=np.uint8),
```

(commented out - they are failed now, but we probably want them to be fixed).